core/src/main/scala/org/apache/spark/storage/BlockObjectWriter.scala

```diff
@@ -117,7 +117,7 @@ private[spark] class DiskBlockObjectWriter(
 
   /** Calling channel.position() to update the write metrics can be a little bit expensive, so we
     * only call it every N writes */
-  private var writesSinceMetricsUpdate = 0
+  private var numRecordsWritten = 0
```

Comment:
It would be good to define this precisely with a comment. I think it's the number of records ever written, excluding reversions (or instead, you could update this if the writes are reverted).